core/engine/src/spanned_source_text.rs

```diff
@@ -0,0 +1,52 @@
+use boa_ast::{LinearSpan, SourceText};
+use boa_gc::{Finalize, Gc, Trace};
+
+#[derive(Trace, Finalize)]
+pub(crate) struct SourceTextInner {
+    #[unsafe_ignore_trace]
+    source_text: SourceText,
+}
+impl SourceTextInner {
+    #[must_use]
+    pub(crate) fn new(code: &mut boa_ast::Script) -> Option<Gc<Self>> {
+        code.take_source()
+            .map(|source_text| Gc::new(Self { source_text }))
+    }
```

Comment:
Just make this function take `boa_ast::SourceText` instead of the `Script`. It does not need anything else. Also together with the other suggestions it would just return `Self`.